pkg/controller/cloneset/update/cloneset_update.go

```diff
@@ -166,16 +174,22 @@ func (c *realControl) updatePod(cs *appsv1alpha1.CloneSet,
 			}
 		}
 
-		inplacing, err := c.inplaceControl.UpdateInPlace(pod, oldRevision, updateRevision)
-		if inplacing && err == nil {
+		opts := &inplaceupdate.UpdateOptions{}
+		if cs.Spec.UpdateStrategy.InPlaceUpdateStrategy != nil {
+			opts.GracePeriodSeconds = cs.Spec.UpdateStrategy.InPlaceUpdateStrategy.GracePeriodSeconds
+		}
+
+		res := c.inplaceControl.Update(pod, oldRevision, updateRevision, opts)
+
+		if res.InPlaceUpdate && res.UpdateErr == nil {
 			c.recorder.Eventf(cs, v1.EventTypeNormal, "SuccessfulUpdatePodInPlace", "successfully update pod %s in-place", pod.Name)
 			c.updateExp.ExpectUpdated(clonesetutils.GetControllerKey(cs), updateRevision.Name, pod)
```

Comment:
I am a little confused here. If the pod is in graceperiod and the expectUpdate is set for pod, will the next reconcile be blocked since the expectation is not satisfied because the pod is not updated yet.

Reply:
The same reason, those pods in graceperiod have already been updated to the latest revision-hash in labels, so it will be satisfied in next reconcile.